tools/wake-unit/rope.cpp

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2022 SiFive, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You should have received a copy of LICENSE.Apache2 along with
+ * this software. If not, you may obtain a copy at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <wcl/rope.h>
+
+#include "unit.h"
+
+TEST(rope_basic) {
+  wcl::rope_builder builder;
+  builder.append("Hello");
+  builder.append(" ");
+  builder.append("World");
+  builder.append("!");
+
+  {
+    wcl::rope_builder other;
+    other.append("My name is");
+    other.append(" John");
+    wcl::rope r = other.build();
+    builder.append(" ");
+    builder.append(r);
+  }
+
+  wcl::rope r = builder.build();
+  std::string expected = "Hello World! My name is John";
+  ASSERT_EQUAL(expected.size(), r.size());
+  ASSERT_EQUAL(expected, r.as_string());
```

Comment:
These can both be EXPECT so that you get more information by both being displayed.